src/components/OpeningHours/OpeningHours.vue

```diff
@@ -0,0 +1,189 @@
+<template>
+  <div class="opening-hours">
+    <button
+      v-if="store.isMobile || showToggleOnDesktop"
+      :aria-label="
+        openingHoursAreVisible ? $t('hideOpeningHours') : $t('showOpeningHours')
+      "
+      class="opening-hours__toggle button"
+      :class="{ 'opening-hours__toggle--open': openingHoursAreVisible }"
+      @click="toggleOpeningHours"
+    >
+      <SvgIcon name="back-icon" class="opening-hours__toggle-icon" />
+
+      <span class="opening-hours__toggle-label">
+        {{ $t("openingHours") }}
+      </span>
+    </button>
+
+    <div
+      v-if="openingHoursAreVisible || (!store.isMobile && !showToggleOnDesktop)"
+    >
+      <h3 class="opening-hours__title">
+        {{ $t("comingWeek") }}
+      </h3>
+
+      <dl class="opening-hours__overview">
+        <template v-for="(timeSlot, index) in timeSlots" :key="index">
+          <dt class="opening-hours__day">
+            {{ index === 0 ? $t("today") : $t(timeSlot.day) }}
+          </dt>
+          <dd class="opening-hours__time">
+            <template v-if="timeSlot.times.length">
+              <p
+                v-for="(time, timeIndex) in timeSlot.times"
+                :key="timeIndex"
+                :class="{
+                  'opening-hours__busy': time.busy,
+                }"
+              >
+                {{ renderTime(time.time[0]) }} - {{ renderTime(time.time[1]) }}
+              </p>
+            </template>
+            <template v-else>
+              {{ $t("closed") }}
+            </template>
+          </dd>
+        </template>
+      </dl>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { useStore } from "~/stores/store";
+import type { OpeningHours } from "~/types/OpeningHours";
+
+export interface Props {
+  openingHoursBuilding: OpeningHours[];
+  openingHoursSpace: OpeningHours[];
+  showToggleOnDesktop: boolean;
+}
+const props = withDefaults(defineProps<Props>(), { showToggleOnDesktop: true });
+const openingHoursAreVisible = ref(false);
+const store = useStore();
+
+function toggleOpeningHours() {
+  openingHoursAreVisible.value = !openingHoursAreVisible.value;
+}
+
+function renderTime(dateStamp: string) {
+  const date = new Date(dateStamp);
+  const time = date.toLocaleString("nl-NL", {
+    hour: "numeric",
+    minute: "2-digit",
+  });
+  return time;
+}
+
+const timeSlots = computed(() => {
+  return props.openingHoursSpace.map((openingHour, dayIndex) => {
+    const buildingHoursThisDay = props.openingHoursBuilding[dayIndex].time;
+
+    const buildingStart = buildingHoursThisDay.length
+      ? buildingHoursThisDay[buildingHoursThisDay.length - 1][0]
+      : ""; //TODO: it was [] before and I do not know what it means. Both should throw!
```

Comment:
I think this is good as a fix, because I think the logic above returns a time string otherwise, like "08:00" or something.

Reply:
Yeah, sure, TypeScript agrees on that. An empty string for a Date constructor doesn't work any better than an empty array, but at least it's less scary type wise